### ISSUE TITLE
Images don't display in the information modal window for the protected folder

### ISSUE THREAD

**Issue by Evgenioni** [contao/core#7795]
If the folder is protected, users can edit the file and the image, the image is displayed in list view, but in the information modal window it's not.
![image 1](https://cloud.githubusercontent.com/assets/2331446/7440863/eab28616-f0ee-11e4-8aca-8bcb046dba60.png)
Checked in Contao 3.4.5 Official Demo

**Comment by leofeyer** [comment]
This is a bit tricky. Of course, we could copy the file to the `/assets/images` folder, but that would also imply that the copy is accessible for everyone. However, since the folder is protected, this is probably not what we want.

@contao/developers /cc

**Comment by Toflar** [comment]
Why not just piping the image like in the downloads element?

**Comment by leofeyer** [comment]
What do you mean by "piping"? We are creating thumbnail images in the downloads element, too. The difference is that its thumbnails in the download elements but full images in the file manager popup.

**Comment by Toflar** [comment]
I don't get the issue then. I thought it's just about the missing preview because the http request is blocked by apache.

**Comment by leofeyer** [comment]
It is about images which you don't want to be publicly available (e.g. because you are selling them).

It is one thing to create a small 70x50 pixel icon of this protected image; if someone finds the thumbnail, they do not have the original.

But the back end popup shows the image in full size, which means there would be a publicly available copy of the full image somewhere in `/assets/images`. If someone finds this copy, they **do** have the original image.

**Comment by Toflar** [comment]
Okay, I understood it correctly then. So I stick with my initial comment: why not just pipe the original image and never crop it?

**Comment by leofeyer** [comment]
What do you mean by "pipe"? Adding the image via the `data` attribute?

**Comment by Toflar** [comment]
Basically like for the download element where we pipe to check the permissions. Easy illustration:

```php
$file = file_get_contents('http://xy.de/contao/main.php?imagePreview=path/of/original/file.png');
```

This means the file `file.png` can be accessed when logged in even though it's protected over http.

**Comment by leofeyer** [comment]
But the browser cannot handle the raw output of `file_get_contents()`. We either have to store it in a file or add it inline via `data:image/*`, haven't we?

**Comment by ausi** [comment]
IMO he means accessing the image via a request to PHP, e.g. 

```html
<img src="http://xy.de/contao/main.php?imagePreview=path/of/original/file.png">
```

Like here: [system/modules/core/elements/ContentDownload.php:66](https://github.com/contao/core/blob/8f678aaf08102432f943241a90d13c1c308741bd/system/modules/core/elements/ContentDownload.php#L66)

**Comment by leofeyer** [comment]
Ah, now I get it! @Evgenioni can you please check if the following change in the `be_popup` template works for you?

```php
<?php if ($this->isImage): ?>
  <div id="preview">
    <div class="checkerbox" style="width:<?= $this->width ?>px;height:<?= $this->height ?>px">
      <img src="<?= $this->href ?>" width="<?= $this->width ?>" height="<?= $this->height ?>" alt="">
    </div>
  </div>
<?php endif; ?>
```

The `src` attribute now uses `$this->href` instead of `$this->src`.

**Comment by aschempp** [comment]
How about adding a base64 inline image instead?

**Comment by leofeyer** [comment]
Why? It is a lot more work and it has no advantages.

**Comment by aschempp** [comment]
Dunno, just a quick idea how to transfer the protected data. Should be fairly simple. You’re aware the `sendFileToBrowser` won’t work? ;-)

**Comment by leofeyer** [comment]
It does work on my system.

**Comment by Evgenioni** [comment]
@leofeyer Yes, it works fine. Image doesn't have a copy in assets/images.

**Comment by Evgenioni** [comment]
My system: ubuntu server 14.04, apache 2.2+nginx 1.6, memcached 1.2, php 5.4
and home: ubuntu server 14.04. apache 2.4 (+nginx 1.7), memcached 1.4, php 5.6

**Comment by leofeyer** [comment]
Fixed in e78e805c41641da2b1aa1b6b8d75aa31de3bc98e.

**Comment by Zeromax** [comment]
@leofeyer will this Change make it to contao 3.2?
